lib/src/test/java/io/ably/lib/test/rest/RestAuthTest.java

```diff
@@ -461,5 +469,240 @@ public Object getTokenRequest(TokenParams params) throws AblyException {
 		}
 	}
 
+	/**
+	 * Verify token details has null client id after authenticating with null client id,
+	 * the message gets published, and published message also does not contain a client id.<br>
+	 * <br>
+	 * Spec: RSA8f1
+	 */
+	@Test
+	public void auth_clientid_null_success() {
+		try {
+			final TestVars testVars = Setup.getTestVars();
+
+			// implement callback, using Ably instance with key
```

Comment:
I couldn't see the difference besides spec reference which I added upon @mattheworiordan 's request. It would be really helpful for me, if you state some differences.